internal/service/advancedclustertpf/validate_schema.go

```diff
@@ -0,0 +1,77 @@
+package advancedclustertpf
```

Comment:
This approach differs slightly from legacy implementation but I argue it is better:
1. Supports raising errors during planning
2. Reusable
3. Easy to test, see resource_schema_test.go